CMakeLists.txt

```diff
@@ -59,7 +59,26 @@ else()
 endif()
 
 target_include_directories(gfx PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(gfx PRIVATE d3d12 dxgi)
+# Find the latest windows SDK needed for d3d12 libs
+if(NOT CMAKE_WINDOWS_KITS_10_DIR)
+    set(CMAKE_WINDOWS_KITS_10_DIR "C:/Program Files (x86)/Windows Kits/10")
+endif()
+if(CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION)
+    STRING(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+" WINDOWS_KIT_VERSION ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION})
+else()
+    set(WINDOWS_KIT_VERSION "10.0.0.0")
+    file(GLOB _KITS RELATIVE ${CMAKE_WINDOWS_KITS_10_DIR}/Lib ${CMAKE_WINDOWS_KITS_10_DIR}/Lib/*)
+    foreach(_KIT ${_KITS})
+        if(${_KIT} VERSION_GREATER_EQUAL ${WINDOWS_KIT_VERSION})
+            set(WINDOWS_KIT_VERSION ${_KIT})
+        endif()
+    endforeach()
```

Comment:
When not building for VS we dont have a target version so instead search for the highest version available